tendrl/node_agent/manager/manager.py

```diff
@@ -239,6 +246,31 @@ def on_pull(self, raw_data):
                     self.etcd_client.write(("nodes/%s/Disks/free/%s") % (
                         raw_data["node_id"], disk), "")
 
+    def load_and_execute_discovery_plugins(self, node_id):
```

Comment:
Having a single manger would be troublesome because all the plugins will be loaded into same array and you would have to run the unnecessary type checks while doing the discoveries, which adds up complexities. I would prefer to have this simple approach of having individual managers per modules